test/test_dtype.py

```diff
@@ -159,7 +159,11 @@ def test_bf16_disk_write_read(self):
 
 class TestHalfDtype(TestDType): DTYPE = dtypes.half
 
-class TestFloatDType(TestDType): DTYPE = dtypes.float
+class TestFloatDType(TestDType):
+  DTYPE = dtypes.float
+  @unittest.skipIf(getenv("HIP", 0) == 1 or getenv("CUDA", 0) == 1 or getenv("METAL", 0) == 1, "doesn't work on CUDA, HIP, or METAL")
```

Comment:
why does it fail on these backends?

Reply:
tbh, I don't have access to any of these devices to research an answer that gives a more specific reason to use, but I noticed that it is a pattern for some casting ops to fail on some backends, so I wrote a skip line initially to make it stand for review.

Reply:
After researching, I found that the issue wasn't a matter of a failing test, but rather implementation-defined behavior. The mentioned devices treat `float` to `uint` differently than the rest. So I reflected that in my last commit.